Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -1212,129 +968,123 @@ private bool MatchesPublishedServerUrl(IPObject source, bool isInExternalSubnet,
-                    _logger.LogDebug("{Source}: GetBindInterface: Using first user defined external interface. {Result}", source, result);
+                    result = NetworkExtensions.FormatIPString(bindAddress);
+                    _logger.LogDebug("{Source}: Internal request received, matching internal bind address found: {Result}", source, result);
                     return true;
                 }
-
-                result = FormatIP6String(addresses[0].Address);
-                _logger.LogDebug("{Source}: GetBindInterface: Selected first user defined interface. {Result}", source, result);
-
-                if (isInExternalSubnet)
-                {
-                    _logger.LogWarning("{Source}: External request received, however, only an internal interface bind found.", source);
-                }
-
-                return true;
             }
 
             return false;
         }
 
         /// <summary>
-        /// Attempts to match the source against an external interface.
+        /// Attempts to match the source against external interfaces.
         /// </summary>
         /// <param name="source">IP source address to use.</param>
         /// <param name="result">The result, if a match is found.</param>
         /// <returns><c>true</c> if a match is found, <c>false</c> otherwise.</returns>
-        private bool MatchesExternalInterface(IPObject source, out string result)
+        private bool MatchesExternalInterface(IPAddress source, out string result)
         {
-            result = string.Empty;
-            // Get the first WAN interface address that isn't a loopback.
-            var extResult = _interfaceAddresses
-                .Exclude(_bindExclusions, false)
-                .Where(p => !IsInLocalNetwork(p))
-                .OrderBy(p => p.Tag)
-                .ToList();
+            // Get the first external interface address that isn't a loopback.
+            var extResult = _interfaces.Where(p => !IsInLocalNetwork(p.Address)).OrderBy(x => x.Index).ToArray();
 
-            if (extResult.Any())
+            // No external interface found
+            if (extResult.Length == 0)
             {
-                // Does the request originate in one of the interface subnets?
-                // (For systems with multiple internal network cards, and multiple subnets)
-                foreach (var intf in extResult)
+                result = string.Empty;
+                _logger.LogWarning("{Source}: External request received, but no external interface found. Need to route through internal network.", source);
+                return false;
+            }
+
+            // Does the request originate in one of the interface subnets?
+            // (For systems with multiple network cards and/or multiple subnets)
+            foreach (var intf in extResult)
+            {
+                if (!IsInLocalNetwork(intf.Address) && intf.Subnet.Contains(source))
```

Comment:
The first part will always be true, right? `extResult` only contains interfaces whose addresses are not in LAN.

Reply:
Correct.